src/vsr/replica.zig

```diff
@@ -715,7 +715,10 @@ pub fn ReplicaType(
                 // that SV to a DVC (dropping the hooks), and never finished the view change.
                 if (op_head == null) {
                     assert(self.view > self.log_view);
-                    if (self.journal.op_maximum() < self.op_checkpoint()) {
+                    if (self.journal.op_maximum() < self.op_checkpoint() or
+                        // Corrupted root prepare:
+                        (self.journal.op_maximum() == 0 and self.journal.header_with_op(0) == null))
```

Comment:
We _could_ farther restrict this clause to `op_checkpoint() == 0`, though that is sort of implied by `op_maximum == 0`, and since we didn't exit earlier due to `self.journal.faulty.count == constants.journal_slot_count`.

Reply:
> though that is sort of implied by op_maximum == 0, and since we didn't exit earlier due to self.journal.faulty.count == constants.journal_slot_count.

This follows more directly --- if we get to the second arm of this `or`, we know that `op_maximum >= op_checkpoint`, so checkpoint can only be zero. 

Actually, I think the `faulty=slot_count` implication would be wrong --- our journal is larger than a checkpoint, so, it could be the case that:

* you are one the _first_ checkpoint
* all _real_ prepares are corrupted
* but there are some valid reserved slots still

In this case, our op_maximum would be 0, and we'll enter this branch via `self.journal.op_maximum() < self.op_checkpoint()` condition